src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -946,25 +950,29 @@ void AndroidDeviceControllerWrapper::OnICDRegistrationComplete(chip::NodeId icdN
     jbyteArray jSymmetricKey          = nullptr;
     CHIP_ERROR methodErr =
         JniReferences::GetInstance().FindMethod(env, mJavaObjectRef, "onICDRegistrationComplete",
-                                                "(IJJLchip/devicecontroller/ICDDeviceInfo;)V", &onICDRegistrationCompleteMethod);
+                                                "(ILchip/devicecontroller/ICDDeviceInfo;)V", &onICDRegistrationCompleteMethod);
     VerifyOrReturn(methodErr == CHIP_NO_ERROR,
                    ChipLogError(Controller, "Error finding Java method: %" CHIP_ERROR_FORMAT, methodErr.Format()));
 
     methodErr = chip::JniReferences::GetInstance().GetClassRef(env, "chip/devicecontroller/ICDDeviceInfo", icdDeviceInfoClass);
     VerifyOrReturn(methodErr == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find class ICDDeviceInfo"));
 
-    icdDeviceInfoStructCtor = env->GetMethodID(icdDeviceInfoClass, "<init>", "([B)V");
+    icdDeviceInfoStructCtor = env->GetMethodID(icdDeviceInfoClass, "<init>", "([BILjava/lang/String;JJJ)V");
     VerifyOrReturn(icdDeviceInfoStructCtor != nullptr, ChipLogError(Controller, "Could not find ICDDeviceInfo constructor"));
 
     methodErr =
         JniReferences::GetInstance().N2J_ByteArray(env, symmetricKey.data(), static_cast<jint>(symmetricKey.size()), jSymmetricKey);
     VerifyOrReturn(methodErr == CHIP_NO_ERROR,
                    ChipLogError(Controller, "Error Parsing Symmetric Key: %" CHIP_ERROR_FORMAT, methodErr.Format()));
 
-    icdDeviceInfoObj = env->NewObject(icdDeviceInfoClass, icdDeviceInfoStructCtor, jSymmetricKey);
+    UtfString jUserActiveModeTriggerInstruction(env, mUserActiveModeTriggerInstruction);
```

Comment:
UtfString is problematic, which is creating reference in local jni reference table and global jni reference table, and is leaking. I think we should directly use NewStringUTF.

jstring localRef = mEnv->NewStringUTF(mUserActiveModeTriggerInstruction) maybe it is better

Reply:
https://github.com/project-chip/connectedhomeip/pull/31193/files#r1447675267
I have not yet pointed out the issue in you previous PR about the potential reference leak. please address in this PR, thanks

Reply:
Oh.. I missed this code. I will modify this.